pkg/controller/nexus/resource/validation/validation.go

```diff
@@ -184,6 +187,43 @@ func (v *Validator) setProbeDefaults(nexus *v1alpha1.Nexus) {
 	}
 }
 
+// must be called only after image defaults have been set
+func (v *Validator) setUpdateDefaults(nexus *v1alpha1.Nexus) {
+	if nexus.Spec.AutomaticUpdate.Disabled {
+		return
+	}
+
+	image := strings.Split(nexus.Spec.Image, ":")[0]
+	if image != NexusCommunityImage {
+		log.Warnf("Automatic Updates are enabled, but 'spec.image' is not using the community image (%s). Disabling automatic updates", NexusCommunityImage)
+		nexus.Spec.AutomaticUpdate.Disabled = true
+		return
+	}
+
+	if nexus.Spec.AutomaticUpdate.MinorVersion == nil {
+		log.Debugf("Automatic Updates are enabled, but no minor was informed. Fetching the most recent...")
+		minor, err := update.GetLatestMinor()
+		if err != nil {
+			log.Errorf("Unable to fetch the most recent minor. Disabling automatic updates.")
+			nexus.Spec.AutomaticUpdate.Disabled = true
+			return
+		}
+		nexus.Spec.AutomaticUpdate.MinorVersion = &minor
+	}
+
+	log.Debugf("Fetching the latest micro from minor %d", *nexus.Spec.AutomaticUpdate.MinorVersion)
+	tag, ok := update.GetLatestMicro(*nexus.Spec.AutomaticUpdate.MinorVersion)
+	if !ok {
+		// the informed minor doesn't exist, let's just use 'latest'
+		log.Warnf("Latest tag for minor version (%d) not found. Using 'latest' instead and disabling Automatic", *nexus.Spec.AutomaticUpdate.MinorVersion)
```

Comment:
Why not fetching latest available and set `Spec.AutomaticUpdate.MinorVersion=nil` ?

Reply:
Also, I think we could add all those WARNs in form of Events instead.

Reply:
> Why not fetching latest available and set Spec.AutomaticUpdate.MinorVersion=nil ?

One of the objectives of this function is assigning a sane value to `minorVersion` if left uninformed. On line 203 we check for that and put in the latest available minor if left blank by the user.

In that branch of execution (when the user didn't specify a minor) we MUST have a valid map of minor->tag, otherwise we would have gotten a non-nil error on line 206:

```go
203     if nexus.Spec.AutomaticUpdate.MinorVersion == nil {
204         log.Debugf("Automatic Updates are enabled, but no minor was informed. Fetching the most recent...")
205         minor, err := update.GetLatestMinor()
206         if err != nil {
207             log.Errorf("Unable to fetch the most recent minor. Disabling automatic updates.")
208             nexus.Spec.AutomaticUpdate.Disabled = true
209             return
210         }
211         nexus.Spec.AutomaticUpdate.MinorVersion = &minor
212     }
```

So in that scenario `ok` must be equal to `true`.

The other scenario is if the user has specified a minor, but it has not been found. For the most part, hitting this means the CR simply doesn't have a valid minor, so going for the latest minor and keeping automatic updates enabled does make sense here. This can also mean that we couldn't fetch the tags, although the user has informed a valid minor. In that scenario we also would not have the tag for the latest minor, so we couldn't simply use that, meaning we would need to use `latest`.

Does that flow sound good to you @ricardozanini? Try the latest available minor while keeping automatic updates enabled. If that also isn't available, disable updates and use `latest`.

> Also, I think we could add all those WARNs in form of Events instead.

Well, that wouldn't be consistent with how we inform the user of other changes performed during validation. Are you suggesting we raise events for all changes performed by the validation package?

Reply:
> Does that flow sound good to you @ricardozanini? Try the latest available minor while keeping automatic updates enabled. If that also isn't available, disable updates and use latest.

Makes sense.

>  Are you suggesting we raise events for all changes performed by the validation package?

We haven't the `event` feature at the point we created the validation package. So, at the time was OK using WARNs. Since now we have, we should inform the user via Events API/CR Status that something is not right. For this scenario I think add an Event would make sense and also have an update in the CR status.

Reply:
I think a reasonable policy for raising events would be to raise one if it makes changes to the CR that are not simply default setting. This way the user gets warned on multiple fronts that their original CR being modified in ways they might not expect. Sounds good?

Reply:
But maybe this is better in another PR, with events for all situations in which this happens.

Reply:
Yes in another PR would be better, but for this particular scenario in this comment, I think we can change to the event instead since it seems pretty important to inform the user what happened. wdyt?

Reply:
Will do